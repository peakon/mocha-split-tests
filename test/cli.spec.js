"use strict";

const cli = require("../lib/cli");

describe("cli", function() {
  it("parses command line args and distributes tests into groups", function() {
    const argv = [
      "-t",
      "3",
      "-g",
      "2",
      "-r",
      "test/data/fakeSpecs/runtime.log",
      "-f",
      "test/data/fakeSpecs/**/*.spec"
    ];
    expect(cli.run(argv)).toMatchSnapshot();
  });

  it("parses multiple globs", function() {
    const argv = [
      "-t",
      "3",
      "-g",
      "2",
      "-r",
      "test/data/fakeSpecs/runtime.log",
      "-f",
      "test/data/fakeSpecs/*.spec",
      "test/data/fakeSpecs/specs/*.spec"
    ];
    expect(cli.run(argv)).toMatchSnapshot();
  });
});
