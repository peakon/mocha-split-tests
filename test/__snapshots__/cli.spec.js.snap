// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli parses command line args and distributes tests into groups 1`] = `"test/data/fakeSpecs/2.spec test/data/fakeSpecs/specs/1.spec"`;

exports[`cli parses multiple globs 1`] = `"test/data/fakeSpecs/4.spec test/data/fakeSpecs/3.spec test/data/fakeSpecs/2.spec test/data/fakeSpecs/specs/1.spec"`;
